src/index.ts

```diff
@@ -442,30 +467,37 @@ class threads {
     );
   };
 
+  @muxrpc('async')
+  public hashtagCount = (
+    opts: Omit<HashtagOpts, 'maxThreadSize' | 'reverse' | 'following'>,
+    cb: CB<number>,
+  ) => {
+    pull(
+      this.rootMsgIdForHashtagMatch(opts.hashtag, {
+        needsDescending: false,
+        msgPassesFilter: makePassesFilter(opts),
+        queryFilter: makeFilterOperator(opts),
+      }),
+      pull.reduce((count: number) => count + 1, 0, cb),
+    );
+  };
+
   @muxrpc('source')
-  public hashtagSummary = (opts: Omit<HashtagOpts, 'threadMaxSize'>) => {
-    const needsDescending = opts.reverse ?? true;
+  public hashtagSummary = (
+    opts: Omit<HashtagOpts, 'threadMaxSize' | 'following'>,
+  ) => {
     const filterOperator = makeFilterOperator(opts);
-    const passesFilter = makePassesFilter(opts);
     const timestamps = new Map<MsgId, number>();
 
     return pull(
-      this.ssb.db.query(
-        where(and(isPublic(), hasHashtag(opts.hashtag), filterOperator)),
-        needsDescending ? descending() : null,
-        batch(BATCH_SIZE),
-        toPullStream(),
-      ),
+      this.rootMsgIdForHashtagMatch(opts.hashtag, {
+        needsDescending: opts.reverse ?? true,
+        msgPassesFilter: makePassesFilter(opts),
+        queryFilter: filterOperator,
+      }),
       pull.through((msg: Msg) =>
         timestamps.set(getRootMsgId(msg), getTimestamp(msg)),
       ),
```

Comment:
~~This `pull.through` is important to get right, it defines the "green timestamp" in Manyverse when a thread has an update, and with this change, I think we broke the logic. Our tests didn't capture that, so we have a hole in our tests too.~~

~~Important to realize here is that (before this PR) `msg` could be any message, either a root post or a reply post. Typically it's a reply post, so what we're doing is saying that the thread's timestamp is~~ **WAIT A MINUTE**

This logic seems wrong before this PR. Maybe it's a leftover from older code that was correct. There's no point to `.set` many times, overwriting the previous value. In fact there seems to be no point in *reading* the `timestamps` Map because we could just pluck the timestamp of the root msg instead.

Huh. I guess I'll refactor this code after merging.